torch/_inductor/mkldnn.py

```diff
@@ -510,21 +513,50 @@ def mkldnn_fuse_fx(gm: torch.fx.GraphModule, example_inputs):
     return gm
 
 
+def create_unary_module(node: torch.fx.node):
+    assert node.op == "call_function", "The current node should be a function node"
+    unary_map = {
+        F.relu: nn.ReLU,
+        F.sigmoid: nn.Sigmoid,
+        F.tanh: nn.Tanh,
+        F.hardswish: nn.Hardswish,
+        F.hardsigmoid: nn.Hardsigmoid,
+        F.leaky_relu: nn.LeakyReLU,
+        F.hardtanh: nn.Hardtanh,
+        F.gelu: nn.GELU,
+        F.relu6: nn.ReLU6,
+        F.silu: nn.SiLU,
+    }
+    return unary_map[node.target](*(node.args[1:]), **(node.kwargs))
+
+
 def fuse_unary(gm: torch.fx.GraphModule):
     modules = dict(gm.named_modules())
 
-    for (unary_module, _), (computation_module, fuse_func,) in itertools.product(
-        unary_modules_map.items(), computation_op_unary_op_fusion_map.items()
-    ):
-        pattern = (computation_module, unary_module)
+    for unary_op, (
+        computation_module,
+        fuse_func,
+    ) in itertools.product(unary_ops, computation_op_unary_op_fusion_map.items()):
+        pattern = (computation_module, unary_op)
         for node in gm.graph.nodes:
-            if matches_module_pattern(pattern, node, modules):
+            if matches_module_pattern(
+                pattern, node, modules
+            ) or matches_module_function_pattern(pattern, node, modules):
                 if (
                     len(node.args[0].users) > 1
                 ):  # Output of computation_node is used by other nodes
                     continue
                 computation_node = modules[node.args[0].target]
-                unary_node = modules[node.target]
+                if node.op == "call_function":
+                    # make sure unary function's inputs only one fx.node(others should be constant value).
+                    if any(isinstance(v, torch.fx.Node) for v in node.args[1:]) or any(
+                        isinstance(v, torch.fx.Node) for _, v in node.kwargs.items()
+                    ):
+                        continue
+                    unary_node = create_unary_module(node)
+                    unary_node.eval()
```

Comment:
Not sure if it will ever happen, but do we need to check if `node` is in `training` before calling `eval()` here?

Reply:
Good catch. I guess it is needed. A similar check was applied after the code at here too: 
https://github.com/pytorch/pytorch/pull/91609/files#diff-7c920a79e1ea5c56baef3ef14306bc2a3a4eb4a13ca7689d93e0549eccd816d4R560

Reply:
For torch function, there doesn't have **training ** attr, even for **nn.module** case of that activation, the **training ** attr is not working, so I think we don't need to do such a check here.